src/ring_theory/ideal/local_ring.lean

```diff
@@ -371,6 +371,32 @@ lemma map_comp (f : T →+* R) (g : R →+* S) [is_local_ring_hom f] [is_local_r
   (local_ring.residue_field.map g).comp (local_ring.residue_field.map f) :=
 ideal.quotient.ring_hom_ext $ ring_hom.ext $ λx, rfl
 
+section
+
+open ring_equiv local_ring
+
+@[simp]lemma map_id_apply {x : residue_field R} : map (ring_hom.id R) x = x :=
+by simp only [map_id, ring_hom.id_apply]
+
+@[simp]lemma map_comp_apply (f : R →+* S) (g : S →+* T) (x : residue_field R)
+[is_local_ring_hom f] [is_local_ring_hom g] :
+map (g.comp f) x = (map g) ((map f) x) :=
```

Comment:
I think this would be  better as
```suggestion
  map g (map f x) = map (g.comp f) x :=
```
With the lemma name changed to `map_map`

Reply:
I changed the name and it works. However, changing to   ```map g (map f x) = map (g.comp f) x :=``` doesn't compile.
What's wrong with that?

Reply:
I'll be able to help once the build falls and there's an error message

Reply:
Alright, I pushed the code

Reply:
When we fix this, I will need to add ```local_ring.residue_field.map_aut``` from #17086

Reply:
This change also seems to cause  `local_ring.residue_field.map_aut` to break. How could do I fix this too?